Makefile

```diff
@@ -289,8 +292,7 @@ apis/anchore/swagger.yaml:
 generate-anchore-client: apis/anchore/swagger.yaml ## Generate client from Anchore OpenAPI spec
 	$(call back_up_file,.gen/anchore/BUILD)
 	$(call generate_openapi_client,apis/anchore/swagger.yaml,anchore,.gen/anchore)
-	sed -i '' 's/whitelist_ids,omitempty/whitelist_ids/' .gen/anchore/model_mapping_rule.go
-	sed -i '' 's/params,omitempty/params/' .gen/anchore/model_policy_rule.go
+	@ if [[ "$$OSTYPE" == "linux-gnu" ]] ; then sed -i 's/whitelist_ids,omitempty/whitelist_ids/' .gen/anchore/model_mapping_rule.go ; sed -i 's/params,omitempty/params/' .gen/anchore/model_policy_rule.go ; else sed -i '' 's/whitelist_ids,omitempty/whitelist_ids/' .gen/anchore/model_mapping_rule.go ; sed -i '' 's/params,omitempty/params/' .gen/anchore/model_policy_rule.go ; fi
```

Comment:
```suggestion
	@ if sed --version &>/dev/null ; then sed -i 's/whitelist_ids,omitempty/whitelist_ids/' .gen/anchore/model_mapping_rule.go ; sed -i 's/params,omitempty/params/' .gen/anchore/model_policy_rule.go ; else sed -i '' 's/whitelist_ids,omitempty/whitelist_ids/' .gen/anchore/model_mapping_rule.go ; sed -i '' 's/params,omitempty/params/' .gen/anchore/model_policy_rule.go ; fi
```

Reply:
If you insist, I'm willing to use your version, but it kind of bugs that it doesn't convey clear information on why the conditional branching is required, it conceals the connection between the macOS/Linux sed flag differences.

If I'm seeing this code for the first time, I'm going to have a hard time understanding how the --version flag correlates to using the in-place flag two different way.

Reply:
See the other suggestion https://github.com/banzaicloud/pipeline/pull/3139#discussion_r479152479

Reply:
the intent behind this approach was that `--version` is a gnu addition just like the optional argument of `-i`

Reply:
Yeah, I understood it, but it is more of an implicit GNU-specific thing than the OS type which means it's more difficult to understand for a newcomer who meets the issue here the first time.